src/transaction/transaction.rs

```diff
@@ -710,7 +712,11 @@ impl<PdC: PdClient> Transaction<PdC> {
 
         let heartbeat_task = async move {
             loop {
-                tokio::time::sleep(DEFAULT_HEARTBEAT_INTERVAL).await;
+                let heartbeat_interval = (|| {
+                    fail_point!("reset-heartbeat-interval", |_| Duration::from_millis(1000));
```

Comment:
Why have this fail point?

Reply:
Because the default heart-interval is 10s and the test will failed if the ttl is 3s. So maybe we need to have a fail point here to set the heartbeat to 1s or make it configurable.

Reply:
Yeah, lets make it configurable, it seems like a weird use of a failpoint to me